qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -15,221 +15,25 @@
-                    SeriesDef(
-                        fit_func=lambda x, p0, p1, p2:
-                            exponential_decay(x, amp=p0, lamb=p1, baseline=p2),
-                    ),
-                ]
-
-                __fixed_parameters__ = ["p1"]
-
-        You can add arbitrary number of parameters to the class variable
-        ``__fixed_parameters__`` from the fit function arguments.
-        This parameter should be defined with the fit functions otherwise the analysis
-        instance cannot be created. In above example, parameter ``p1`` should be also
-        defined in the analysis options. This parameter will be excluded from the fit parameters
-        and thus will not appear in the analysis result.
-
-    Notes:
-        This CurveAnalysis class provides several private methods that subclasses can override.
-
-        - Customize pre-data processing:
-            Override :meth:`~self._format_data`. For example, here you can apply smoothing
-            to y values, remove outlier, or apply filter function to the data.
-            By default, data is sorted by x values and the measured values at the same
-            x value are averaged.
-
-        - Create extra data from fit result:
-            Override :meth:`~self._extra_database_entry`. You need to return a list of
-            :class:`~qiskit_experiments.framework.analysis_result_data.AnalysisResultData`
-            object. This returns an empty list by default.
-
-        - Customize fit quality evaluation:
-            Override :meth:`~self._evaluate_quality`. This value will be shown in the
-            database. You can determine the quality represented by the predefined string
-            "good" or "bad" based on fit result,
-            such as parameter uncertainty and reduced chi-squared value.
-            This returns ``None`` by default. This means evaluation is not performed.
-
-        - Customize fitting options:
-            Override :meth:`~self._generate_fit_guesses`. For example, here you can
-            calculate initial guess from experiment data and setup fitter options.
+from qiskit_experiments.exceptions import AnalysisError
+from qiskit_experiments.framework import ExperimentData, AnalysisResultData, AnalysisConfig
+from qiskit_experiments.warnings import deprecated_function
 
-        See docstring of each method for more details.
+from .base_curve_analysis import BaseCurveAnalysis
+from .curve_data import CurveData, SeriesDef
 
-        Note that other private methods are not expected to be overridden.
-        If you forcibly override these methods, the behavior of analysis logic is not well tested
-        and we cannot guarantee it works as expected (you may suffer from bugs).
-        Instead, you can open an issue in qiskit-experiment github to upgrade this class
-        with proper unittest framework.
 
-        https://github.com/Qiskit/qiskit-experiments/issues
-    """
+class CurveAnalysis(BaseCurveAnalysis):
```

Comment:
Is there any reason to keep this as a "base class" anymore now that there is a true `BaseCurveAnalysisClass`? It looks to me that this could be a directly used analysis class with 1 small change of passing the `series` defs as an `__init__` arg, rather than storing them as a class attribute, and all the subclasses would pass their pre-defined series through the super init method.

Reply:
In light of this, I think this class should also be documented as if it were a proper analysis class, currently all the API docs have been moved to the base class, however we still expect people to subclass this class, not the base class directly, so relevant user/developer API docs should be here as well.

Reply:
As we discussed offline moving series to instance variable is problematic due to picking of lambda functions there. This should be addressed in next release. Regarding the method doc, I readd them in https://github.com/Qiskit/qiskit-experiments/pull/765/commits/ef9e61b55d8adfb73d077860533e030e3f4c58f3

Reply:
That's a good point, so lets think about it in more detail and discuss possible options after the release